### PR TITLE
Bug Fix: The value in the setvar should be able to start with - or +. 

### PR DESCRIPTION
**Pull request by soujanyanmbri**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [✅] My code includes positive and negative tests.
- [✅] I have an appropriate description with correct grammar.
- [✅] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [✅] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: internal/actions/setvar.go [hunk 1]
```diff
@@ -150,11 +150,17 @@ func (a *setvarFn) evaluateTxCollection(r plugintypes.RuleMetadata, tx plugintyp
 		if len(value) > 1 {
 			val, err = strconv.Atoi(value[1:])
 			if err != nil {
-				tx.DebugLogger().Error().
-					Str("var_value", value).
-					Int("rule_id", r.ID()).
-					Err(err).
-					Msg("Invalid value")
+				// If the variable doesn't exist, we would need to raise an error. Otherwise, it should be the same value.
+				if strings.HasPrefix(value[1:], "tx.") {
```
Comment (jcchavezs) [inline internal/actions/setvar.go:154]: shouldn't this be `%{tx.`? instead of `tx.`

**Comment by jcchavezs** [comment]
thanks, please update this line https://github.com/corazawaf/coraza/pull/1125/files#diff-4f480854f968ff92545ff3bc611fb34a15a657f6caf11dc203c64b76c0861827R148 with the appropriate comment.

File: internal/actions/setvar_test.go [hunk 2]
```diff
@@ -96,6 +96,16 @@ func TestSetvarEvaluate(t *testing.T) {
 			init:                     "TX.newvar=-%{tx.missingvar}",
 			expectInvalidSyntaxError: true,
 		},
+		{
+			name:                     "Non Numerical Operation - If the value starts with -",
+			init:                     "TX.newvar=----expected_value",
+			expectInvalidSyntaxError: false,
+		},
```
Comment (jcchavezs) [inline internal/actions/setvar_test.go:103]: should we include `expectNewVarValue` in the test swell?

File: internal/actions/setvar.go (hunk 1 above)
Comment (soujanyanmbri) [reply internal/actions/setvar.go:154]: It is either resolve to the value or stays as tx.missingValue.

Comment (jcchavezs) [reply internal/actions/setvar.go:154]: But according to https://github.com/corazawaf/coraza/pull/1125/files#diff-488435c221aebedf4137cb98424234aab81decfd749827ab96e188c53bd53c9fR90-R92 it returns a syntax error.

Comment (soujanyanmbri) [reply internal/actions/setvar.go:154]: Here's the value: Screenshot 2024-08-05 at 2 38 24 PM

File: internal/actions/setvar_test.go (hunk 2 above)
Comment (soujanyanmbri) [reply internal/actions/setvar_test.go:104]: Done

File: internal/actions/setvar.go (hunk 1 above)
Comment (jcchavezs) [reply internal/actions/setvar.go:155]: I see. It is still not clear to me how we should deal with the interpolation error in this code path. Let me check it.

Comment (soujanyanmbri) [reply internal/actions/setvar.go:155]: Sure, thanks!

Comment (soujanyanmbri) [reply internal/actions/setvar.go:155]: Hey @jcchavezs , 
Any update on how this error could be handled? 

Thanks!

**Comment by jcchavezs** [comment]
Sorry for the delay. Prioritizing this for now.

File: internal/actions/setvar.go (hunk 1 above)
Comment (jcchavezs) [reply internal/actions/setvar.go:155]: Ok I found the issue, when we compile a macro e.g. `%{tx.var}`, the text for the macro becomes `tx.var` and not `%{tx.var}` and when the key does not exist we return the `text` which in this case is `tx.var`. I will discuss that over another PR.

**Review by jcchavezs — APPROVED** [review]

**Pull request merged**